libparsec/crates/client_connection/src/authenticated_cmds/mod.rs

```diff
@@ -206,6 +206,13 @@ impl AuthenticatedCmds {
         // No Content-Type as this request is a GET
         content_headers.insert(CONTENT_LENGTH, HeaderValue::from_static("0"));
         content_headers.insert(ACCEPT, HeaderValue::from_static(EVENT_STREAM_CONTENT_TYPE));
+        if let Some(id) = last_event_id {
+            content_headers.insert(
+                "last-event-id",
+                HeaderValue::from_str(id)
+                    .expect("Invalid char in `id` to be converted into a header value"),
```

Comment:
Is it safe ?

Reply:
panic-ing is safe, yes :nerd_face:

Reply:
I've choose to use a panic because I was lazy to create a new variant in `ConnectionError` for an invalid id provided by the server.

Should I add a variant nevertheless for that ?

Reply:
> I've choose to use a panic because I was lazy to create a new variant in ConnectionError for an invalid id provided by the server.

As a generic rule, it shouldn't be possible for a peer to make the application panic. Otherwise this typically opens the door to denial of service attacks.

In the current case, we should do
- validate `last-event-id` as soon as possible, then pass it around as dedicated `SSEEventID` type (typically something that just wrap `String` and implement a `FromStr` that do the validation and a `AsRef<str>` to access it).
- if validation fail, we end up with a `ConnectionError::InvalidLastEventID`, so that it can be handle like a regular error
- in `sse_request_builder` we take as parameter the `SSEEventID`, and hence can have a `.expect("Already validated")` ;-)

Reply:
done